docs/babelconfigjs.md

```diff
@@ -0,0 +1,97 @@
+---
+title: babel.config.js
+id: babelconfigjs
+---
+
+## Lookup behavior
+
+The lookup behavior is direct as Babel will only look for a `babel.config.js` in the root directory (where your `package.json` is by default). If configuration is needed, see [specifying the root folder](babelconfigjs.md#specifying-the-root-folder).
+
+## Merging behavior
+
+Unlike other formats (see the [configuration guide](configuration.md)), Babel won't try to merge configurations.
+
+> Note: the configuration defined in `babel.config.js` is applied to your entire project.
+
+## Conditional configuration
+
+Since it's just a regular JavaScript file, you can write any arbitrary conditions before returning the configuration, for example:
+
+```js
+module.exports = function () {
+  const plugins = [ ... ];
+
+  if (true === false) {
+    plugins.push(...);
+  }
+
+  return { plugins };
+}
+```
+
+### Environment
+
+A conditional configuration is useful when you want to use certain plugins/presets based on the current environment, for example:
```

Comment:
should mention that `"env"` still works fine for people, not sure if it should go here or elsewhere

Reply:
Yes, I wasn't sure if we wanted to include it here. With this format I'd like to promote that you can use `process.env` directly instead, which means that the merging strategy is done on their side. I think it's better to avoid mentioning it. Thoughts?